homeassistant/components/notify/kodi.py

```diff
@@ -0,0 +1,74 @@
+﻿"""
+Kodi notification service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.kodi/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (ATTR_ICON, CONF_HOST, CONF_PORT,
+                                 CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.components.notify import (ATTR_TITLE, ATTR_TITLE_DEFAULT,
+                                             ATTR_DATA, PLATFORM_SCHEMA,
+                                             BaseNotificationService)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['jsonrpc-requests==0.3']
+
+DEFAULT_PORT = 8080
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+})
+
+ATTR_DISPLAYTIME = 'displaytime'
+
+
+def get_service(hass, config):
+    """Return the notify service."""
+    url = '{}:{}'.format(config.get(CONF_HOST), config.get(CONF_PORT))
+
+    auth = (config.get(CONF_USERNAME), config.get(CONF_PASSWORD))
+
+    return KODINotificationService(
+        url,
+        auth
+    )
+
+
+# pylint: disable=too-few-public-methods
+# pylint: disable=broad-except

```

Comment:
We should never disable `broad-except`, we should just not catch broad exceptions. JSONRPC-requests shows clearly which exceptions can be thrown, and we should use that.